src/components/SwapWidget/SwapWidget.tsx

```diff
@@ -758,7 +758,7 @@ const SwapWidget: FC = () => {
 
       <Overlay
         onCloseButtonClick={() => setShowTokenSelectModalFor(null)}
-        isHidden={!showTokenSelectModalFor}
+        isHidden={!showTokenSelectModalFor || !active}
```

Comment:
for both Make and Swap, this is cleaner when the user reconnects, as the tokenModal re-opens in a clunky manner in current setup

```
useEffect(() => {
    if (!active) {
      setShowTokenSelectModalFor(null)
    }
  }, [active])```

Reply:
Thanks. Fixed!